#! /bin/sh /usr/share/dpatch/dpatch-run
## vdr-cap_net_raw.diff.dpatch
##
## Adapted from vdr-cap_net_raw.diff shipped with streamdev source but
## rebased onto vdr-1.7.21.
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Prevent VDR from dropping the CAP_NET_RAW capability required to
## DP: bind an IGMP multicast socket.

@DPATCH@
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' vdr-1.7.21~/vdr.c vdr-1.7.21/vdr.c
--- vdr-1.7.21~/vdr.c	2011-09-21 13:13:32.880472062 +0100
+++ vdr-1.7.21/vdr.c	2011-09-21 13:15:33.639622178 +0100
@@ -116,7 +116,7 @@
 static bool DropCaps(void)
 {
   // drop all capabilities except selected ones
-  cap_t caps = cap_from_text("= cap_sys_nice,cap_sys_time,cap_sys_tty_config=ep");
+  cap_t caps = cap_from_text("= cap_sys_nice,cap_sys_time,cap_sys_tty_config,cap_net_raw=ep");
   if (!caps) {
      fprintf(stderr, "vdr: cap_from_text failed: %s\n", strerror(errno));
      return false;
